src/events/game_listener.go

```diff
@@ -0,0 +1,41 @@
+package events
+
+import (
+	"fmt"
+	"game"
+	"reflect"
+	"renderer"
+)
+
+func GameListener(doneChan chan bool, inputChan chan byte, gridChan chan game.Grid, startedGrid game.Grid) {
+	grid := game.DeepCopyGrid(startedGrid)
+	gridChan <- grid
+	for {
+		action := <-inputChan
+		if action == game.ACTION_QUIT {
+			doneChan <- false
+			return
+		}
+
+		if action == game.ACTION_SHUFFLE {
+			renderer.ClearTerminal()
+			fmt.Println("Shuffling...")
```

Comment:
It should be in the `Render` channel. You should only handle game logic in this channel. But don't spend more than an hour on moving it.